### PR TITLE
Add writeOnly type for calendar

### PR DESCRIPTION
**Pull request by alexanderpuchta**
## Goal
with iOS 17 there are two new methods to request access for calendars.

- `requestWriteOnlyAccessToEvents` -> `.calendarWriteOnly`
- `requestFullAccessToEvents` -> `.calendar`

### CODE REVIEW

**Comment by alexanderpuchta** [comment]
closed because of #321

**Comment by ivanvorobei** [comment]
I think your solution cleaner. So we can move on with it

**Comment by alexanderpuchta** [comment]
i have updated the README file with help of #321 and added available iOS 17 only like suggested in https://github.com/sparrowcode/PermissionsKit/pull/321#discussion_r1338245557 @ivanvorobei

File: Sources/CalendarPermission/CalendarPermission.swift [hunk 1]
```diff
@@ -28,15 +28,31 @@ import Foundation
...
+    // MARK: - Init
+    
+    init(kind: Permission.Kind) {
+        self._kind = kind
+    }
+    
+    open override var kind: Permission.Kind { self._kind }
     open var usageDescriptionKey: String? { "NSCalendarsUsageDescription" }
     open var usageFullAccessDescriptionKey: String? { "NSCalendarsFullAccessUsageDescription" }
     open var usageWriteOnlyAccessDescriptionKey: String? { "NSCalendarsWriteOnlyAccessUsageDescription" }
```
Comment (ostatnicky) [inline Sources/CalendarPermission/CalendarPermission.swift:58]: I think here should be similar to https://github.com/sparrowcode/PermissionsKit/pull/321/files#diff-5d1754e5cc6ca3571a075fa6d9a401fb40120b284bd7f85273401de378d0d6d0R40

```
open var usageDescriptionKey: String? {
         if #available(iOS 17.0, *) {
             switch kind {
                  case .calendar: return "NSCalendarsFullAccessUsageDescription"
                  case .calendarWriteOnly: return "NSCalendarsWriteOnlyAccessUsageDescription"                 
             }
         } else {
             return "NSCalendarsUsageDescription"
         }
     }
```

Comment (ivanvorobei) [reply Sources/CalendarPermission/CalendarPermission.swift:58]: Agree, looks more structed

**Comment by ivanvorobei** [comment]
1. Upgraded usage description key

```swift
open var usageDescriptionKey: String? {
        if #available(iOS 17, *) {
            switch kind {
            case .calendar(let access):
                switch access {
                case .full:
                    return "NSCalendarsFullAccessUsageDescription"
                case .write:
                    return "NSCalendarsWriteOnlyAccessUsageDescription"
                }
            default:
                fatalError()
            }
        } else {
            return "NSCalendarsUsageDescription"
        }
    }
```

2. Updated permission access level to enum. For calendar it's `CalendarAccess`. As well 

3. drop `@available(iOS 17.0, *)` for calendar permission becouse no reason to have it. If dev asking write only it mean he want have full access on iOS 16. 

4. next merged 2 calendar permission to one calendar + access value

```swift
public extension Permission {
    
    static func calendar(access: CalendarAccess) -> CalendarPermission {
        CalendarPermission(kind: .calendar(access: access))
    }
}
```

File: Sources/CalendarPermission/CalendarPermission.swift (hunk 1 above)
Comment (ivanvorobei) [reply Sources/CalendarPermission/CalendarPermission.swift:58]: > Ok, but you didn't change it :D

You so fast! Changes pushed maybe 5 min ago

**Pull request merged**